homework-2/middewares/userMiddlewares.ts

```diff
@@ -0,0 +1,61 @@
+import { Request, Response} from 'express';
+import UserService  from '../services/userService';
+import { User } from "../interfaces/User";
+import { RES_MESSAGES, RES_STATUS_CODES } from "../config/constants";
+import { validationResult } from "express-validator";
+
+interface IdParam {
+    id: string;
+}
+
+interface AutoSuggestReqQuery {
+    loginSubstring: string;
+    limit: number;
+}
+
+export function getAllUsersHandler(req: Request, res: Response): void {
+    res.send(UserService.getAllUsers());
+}
+
+export function getUserByIdHandler(req: Request<IdParam>, res: Response): void {
+    const user = UserService.getUserById(req.params.id);
+
+    if (user) {
+        res.send(user);
+    } else {
+        res.status(RES_STATUS_CODES.NOT_FOUND).send(RES_MESSAGES.NOT_FOUND);
+    }
+}
+
+export function getAutoSuggestHandler(req: Request<unknown, unknown, unknown, AutoSuggestReqQuery>, res: Response): void {
+    const autoSuggests = UserService.getAutoSuggestUsers(req.query.loginSubstring, req.query.limit);
+    res.send(autoSuggests);
+}
+
+export function addUserHandler(req: Request<Record<string, unknown>, unknown, User>, res: Response): void {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+        res.status(RES_STATUS_CODES.BAD_REQUEST).send({ errors: errors.array() });
+    } else {
+        const userId = UserService.addUser(req.body);
+        res.status(RES_STATUS_CODES.CREATED).send(userId);
+    }
+}
+
+export function updateUserHandler(req: Request<unknown, unknown, User>, res: Response): void {
```

Comment:
It is confusing logic. If you cannot found the user please just send not found status

Reply:
Fixed, thanks